### PR TITLE
#160577495 Users should be able to report an article

### PR DESCRIPTION
**Pull request by lukorito**
#### What does this PR do?
This PR creates a feature that enables authenticated users to report an article to the admin
#### Description of Task to be completed?
Have the following endpoint working:
`POST articles/<slug>/report/` Users can report an article using the slug
#### How should this be manually tested?
**Prerequisites**
- Postgres SQL
- Postman

Procedure
- Clone the repository `https://github.com/andela/ah-shakas.git`
- Change your working directory to the project folder `cd ah-shakas`
- Checkout to the current branch `git checkout ft-report-articles-160577495`
- Create a virtual environment using [pipenv](https://pipenv.readthedocs.io/en/latest/) `pipenv install` which will also install the dependencies
- Setup the environment variables as shown in `.env-example` and save it as `.env`
- Use `pipenv shell` on the terminal to access the virtualenvironment and all the variables will be exported
- Run the necessary migrations `python manage.py migrate`
- Run the tests `python manage.py test` to see the passing tests
- Run the Django server with `python manage.py runserver`
- Use [POSTMAN](https://www.getpostman.com/) or an API Development Environment of choice
- Sign up a user using the following payload format at `api/users/`:
```
{
	"user": {
		"username": "username",
		"email": "email@email.com",
		"password": "Password4"
	}
}
```
- Login the user using the above payload at `api/user/login` and you will receive a token
- Use the format `key: Authorization, value:Token` for the header
- Create an article by accessing `api/articles/` and sending a payload in the following format:
```
{
	"article" : {
		"title": "This is a title",
		"description": "This is a description",
		"body": "this is a body",
		"tags": ["sample","tags"]
	}
}
```
- Create another user using the procedure above with a different `username` and `email`. As below
```
{
	"user" : {
		"username": "reporting_user",
                "password": "reporting134",
                "email": "reporting@test.com"
	}
}

```
- Login with the second user and use the token from the second user  Use the format `key: Authorization, value:Token` for the header
- Send a `POST` request to the url `api/articles/{slug}/report/` to report the article with  the following payload:
```
{
	"report" : {
		"report_msg": "I don't like this article because"
	}
}
```
- The following response is expected : 
`{"message": 'An email has been sent to the admin with your request'}`

#### Any background context you want to provide?
As an authenticated user I should be able to report articles that violate terms of the agreement or are plagiarised to the authors haven admin for removal and or any other actions as per the terms of engagement
#### What are the relevant pivotal tracker stories?
[#160577495](https://www.pivotaltracker.com/story/show/160577495)
#### Screenshots (if appropriate)
**Reporting an article**
screenshot 2018-10-21 10 38 45
**Sample email received by Admin**
screenshot 2018-10-21 10 39 48

### CODE REVIEW

**Comment by IncredibleBlackMan** [comment]
Hey, @lukorito thumbs up for a job well done. However, I believe your code would be more readable if you include docstrings and comments.

Also for clarification, what happens in the event that I want to retract my report, probably I left my machine unattended and the cat decided to run the show. Do you think it is a valid proposition to handle deletes? I noticed our code does not handle that.

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -531,3 +537,62 @@ def delete(self, request, slug):
                 , status=status.HTTP_200_OK
             )
 
+
```
Comment (IncredibleBlackMan) [inline authors/apps/articles/views.py]: Add comments and docstrings to this file as has been done in the rest.

**Review by IncredibleBlackMan — DISMISSED** [review]

File: authors/apps/articles/views.py (hunk 1 above)
Comment (lukorito) [reply authors/apps/articles/views.py]: Noted

**Comment by lukorito** [comment]
> Hey, @lukorito thumbs up for a job well done. However, I believe your code would be more readable if you include docstrings and comments.
> 
> Also for clarification, what happens in the event that I want to retract my report, probably I left my machine unattended and the cat decided to run the show. Do you think it is a valid proposition to handle deletes? I noticed our code does not handle that.

@Ramon3162 The process of reporting an article is so lengthy, there are so many instances one can bail out. It is not a direct thing. In my demo, I gave an example of someone reporting. Moreover, the admin will review the article, it does not mean necessarily it will be deleted

**Review by koechkevin — APPROVED** [review]
This looks good to me

**Pull request merged**